modules/33acrossBidAdapter.js

```diff
@@ -137,62 +137,82 @@ function _validateVideo(bid) {
 // NOTE: With regards to gdrp consent data, the server will independently
 // infer the gdpr applicability therefore, setting the default value to false
 function buildRequests(bidRequests, bidderRequest) {
+  const ttxSettings = config.getConfig('ttxSettings');
   const gdprConsent = Object.assign({
     consentString: undefined,
     gdprApplies: false
   }, bidderRequest && bidderRequest.gdprConsent);
-
   const uspConsent = bidderRequest && bidderRequest.uspConsent;
   const pageUrl = (bidderRequest && bidderRequest.refererInfo) ? (bidderRequest.refererInfo.referer) : (undefined);
 
   adapterState.uniqueSiteIds = bidRequests.map(req => req.params.siteId).filter(utils.uniques);
 
-  return bidRequests.map(bidRequest => _createServerRequest(
-    {
-      bidRequest,
-      gdprConsent,
-      uspConsent,
-      pageUrl
-    })
-  );
+  const bidRequestsComplete = bidRequests.map(_inferProduct);
+
+  const enableSRAMode = ttxSettings && ttxSettings.enableSRAMode;
+
+  const keyFunc = (enableSRAMode === 'true') ? _getSRAKey : _getDefaultKey;
+  const groupedRequests = _groupBidRequests(bidRequestsComplete, keyFunc);
+
+  const serverRequests = [];
+
+  for (const key in groupedRequests) {
+    serverRequests.push(
+      _createServerRequest({
+        bidRequests: groupedRequests[key],
+        gdprConsent,
+        uspConsent,
+        pageUrl,
+        ttxSettings
+      })
+    )
+  }
+
+  return serverRequests;
+}
+
+function _groupBidRequests(bidRequests, keyFunc = _getDefaultKey) {
+  const groupedRequests = {};
+
+  bidRequests.forEach(function(req) {
+    const key = keyFunc(req);
+
+    groupedRequests[key] = groupedRequests[key] || [];
+    groupedRequests[key].push(req);
+  });
+
+  return groupedRequests;
+}
+
+function _getSRAKey(bidRequest) {
+  return `${bidRequest.params.siteId}:${bidRequest.params.productId}`;
```

Comment:
nice one! if the pub want to has one request, they need to have the same site id and prod across all the imp. otherwise it will split into different requests